tests/5.0/target/test_target_device.F90

```diff
@@ -0,0 +1,93 @@
+!===--- test_target_device.F90 -------------------------------===//
+!
+! OpenMP API Version 5.0 Nov 2018
+!
+! This test checks the target construct with device clause where device-
+! modifier is either ancestor or device_num. If no device_modifier is 
+! present, the behavior is the same as if device_num were present.
+! 
+!//===----------------------------------------------------------------------===//
+#include "ompvv.F90"
+
+#define N 1024
+
+! Required for device(ancestor: 1)
+!$omp requires reverse_offload
+
+PROGRAM test_target_device
```

Comment:
The requires directive is misplaced – you need to move it into the specification part of the PROGRAM – and all other _compilation units_. Regarding the latter, it also needs to be set in the "ompvv_lib" module. To do so, define "OMPVV_MODULE_REQUIRES_LINE" accordingly.

Note that the meaning of _compilation unit_ differs between Fortran and C/C++:

> _compilation unit_
> For C/C++, a translation unit.
> For Fortran, a program unit.

where "program unit" is defined in Fortran as (here: F2018):
> 3.116
> **program unit**
> main program, external subprogram, module, submodule, or block data program unit (5.2.1).

Namely: Something like the following – it can be also be moved further down in PROGRAM (if that's more readable) – but it must be placed before OMPVV_TEST_OFFLOADING (which is the first exec statement and ends the specification part).

```suggestion
PROGRAM test_target_device
  ! Required for device(ancestor: 1)
  !$omp requires reverse_offload
```

Reply:
Updated.